frontend/src/components/Card.tsx

```diff
@@ -28,165 +27,182 @@ const Card = ({
   social,
   tooltipLabel,
 }: CardProps) => {
-  const [visibleLanguages, setVisibleLanguages] = useState(18)
-  const [visibleTopics, setVisibleTopics] = useState(18)
+  const [visibleLanguages, setVisibleLanguages] = useState(18);
+  const [visibleTopics, setVisibleTopics] = useState(18);
+  const [toggleLanguages, setToggleLanguages] = useState(false);
+  const [toggleTopics, setToggleTopics] = useState(false);
+  const [isMobile, setIsMobile] = useState(false);
 
-  const [toggleLanguages, setToggleLanguages] = useState(true)
-  const [toggleTopics, setToggleTopics] = useState(true)
+  useEffect(() => {
+    const checkMobile = () => {
+      const mobile = window.innerWidth < 768;
+      setIsMobile(mobile);
+      setVisibleLanguages(mobile ? 4 : 18);
```

Comment:
For some reason this is not working as it supposed to. I attached a video recording that shows what I mean. When I do hard-refresh on the page it takes several seconds for this to apply and reduce the number of labels on the card. This is really confusing and it shouldn't be like this for the end user. Can we figure out what could be changed here?

https://github.com/user-attachments/assets/792cc3b1-9a84-4031-876c-0c200ae03465